crypto/src/lib.rs

```diff
@@ -325,6 +293,19 @@ ffi::ffi_item! {
 
 #[ffi_impl_opaque]
 impl PublicKey {
+    /// Creates a new public key from raw bytes received from elsewhere
+    pub fn from_raw(algorithm: Algorithm, payload: ConstVec<u8>) -> Self {
+        Self {
+            digest_function: algorithm,
+            payload,
+        }
+    }
+
+    /// Extracts the raw bytes from public key
+    pub fn into_raw(self) -> (Algorithm, ConstVec<u8>) {
```

Comment:
I don't quite like this API. If we have to have `from_raw` and `into_raw`, rename them to `from_raw_parts` and `into_raw_parts`